lib/libc/sys/ioctl.c

```diff
@@ -88,9 +92,22 @@ _ioctl(int fd, unsigned long com, ...)
 		} else {
 			data = va_arg(ap, void *);
 		}
-		va_end(ap);
 	} else {
-		data = NULL;
+#ifdef __CHERI_PURE_CAPABILITY__
+		if (cheri_getlen((void*)ap) == sizeof(void *))
```

Comment:
Pondering further, I think I'd make this
```suggestion
		if (cheri_getlen((void*)ap) >= sizeof(void *))
```
That might suggest dropping the other cases.